dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -7304,27 +7304,36 @@ void Sedp::match_continue(const GUID_t& writer, const GUID_t& reader)
     }
 
     // Copy reader and writer association data prior to releasing lock
-    DDS::OctetSeq octet_seq_type_info_reader;
-    XTypes::serialize_type_info(*reader_type_info, octet_seq_type_info_reader);
-    const DCPS::ReaderAssociation ra = {
-      *rTls, rTransportContext, reader, *subQos, *drQos,
+    WriterAssociationRecord_rch war;
+    if (call_writer) {
+      war = make_rch<WriterAssociationRecord>(dwr, writer, DCPS::ReaderAssociation());
+      DCPS::ReaderAssociation& ra = const_cast<DCPS::ReaderAssociation&>(war->reader_association_);
```

Comment:
Would it be cleaner just to change this to be a nonconst member?
Or pass it to the constructor (unless that's too slow).

Reply:
I didn't want to go the constructor route because the TransportLocatorSeq is a member of ReaderAssociation (a taoidl-generated type) and unless taoidl's correctly generating move-constructors it would probably wind up copying the sequence twice.

I think having the const is nice for the users of the ReaderAssociationRecord, so I left that in place even though it makes the creation side of things a bit uglier.